### DISCUSSION TITLE
Golang bindings/interoperability story

### DISCUSSION THREAD

**Discussion by tmc** [denoland/deno#8527] [General]
In some contexts it's nice to have js/ts execution capabilities inside a go process. The various v8 go bindings that exist are in a variety of states with regards quality. Having a supported way to interact with libdeno via go would be an interesting bridge.

Could you speak to any roadmapping regarding this or provide some direction for a potential contributor?

Note: I'm aware that this project was initially written in go and then moved to rust.

**Comment by axetroy** [comment]
interesting. I also look forward to it.

Fortunately, ry is also proficient in Golang.

**Comment by kitsonk** [comment]
We have a long road to making the Deno CLI crate consumable as an external API, and even then we are focused on Rust for those bindings.  We have been taking some of the "web" APIs and breaking them out to seperate crates we call "op crates" but that model is still a bit rough and still it is designed for Rust.

From an architectural perspective, [`rusty_v8`](https://github.com/denoland/rusty_v8) are the v8 bindings for Rust.

There is no roadmap to making the whole of the CLI as an externally consumable set of APIs.  Even then there is likely a lot more there than someone would want/need for embedding JavaScript or TypeScript support via v8 in Go.
